pinot-common/src/main/java/com/linkedin/pinot/common/utils/DataTableBuilder.java

```diff
@@ -607,6 +609,30 @@ public static DataSchema fromBytes(byte[] buffer) {
 
     }
 
+    /**
+     * Indicates whether the given {@link DataSchema} is type compatible with this one.
+     * <ul>
+     *   <li>All numbers are type compatible.</li>
+     *   <li>Number is not type compatible with String.</li>
+     *   <li>Single-value is not type compatible with multi-value.</li>
+     * </ul>
+     *
+     * @param anotherDataSchema data schema to compare.
+     * @return whether the two data schemas are type compatible.
+     */
+    public boolean isTypeCompatibleWith(@Nonnull DataSchema anotherDataSchema) {
```

Comment:
The order of columns in data schema should always be same because they are generated by the same broker request.